Mathlib/Order/Basic.lean

```diff
@@ -1261,32 +1261,36 @@ instance [∀ i, Preorder (π i)] [∀ i, DenselyOrdered (π i)] :
           ⟨le_update_iff.2 ⟨ha.le, fun _ _ ↦ le_rfl⟩, i, by rwa [update_self]⟩,
           update_le_iff.2 ⟨hb.le, fun _ _ ↦ hab _⟩, i, by rwa [update_self]⟩⟩
 
-theorem le_of_forall_le_of_dense [LinearOrder α] [DenselyOrdered α] {a₁ a₂ : α}
-    (h : ∀ a, a₂ < a → a₁ ≤ a) : a₁ ≤ a₂ :=
+section LinearOrder
+variable [LinearOrder α] [DenselyOrdered α] {a₁ a₂ : α}
+
+theorem le_of_forall_le_of_dense (h : ∀ a, a₂ < a → a₁ ≤ a) : a₁ ≤ a₂ :=
   le_of_not_gt fun ha ↦
     let ⟨a, ha₁, ha₂⟩ := exists_between ha
     lt_irrefl a <| lt_of_lt_of_le ‹a < a₁› (h _ ‹a₂ < a›)
 
-theorem eq_of_le_of_forall_le_of_dense [LinearOrder α] [DenselyOrdered α] {a₁ a₂ : α} (h₁ : a₂ ≤ a₁)
+lemma forall_gt_imp_ge_iff_le_of_dense : (∀ a, a₂ < a → a₁ ≤ a) ↔ a₁ ≤ a₂ :=
```

Comment:
What about `forall_le_iff_le_of_dense` here, to match `le_of_forall_le_of_dense` as above? That is, since you have `X_of_Y_of_Z`, and this lemma has that one as a single direction, the name `Y_iff_X_of_Z` seems more reasonable.

Reply:
What really bothers me with the name `le_of_forall_le_of_dense` is that it doesn't contain `lt`, which is the main difference with the analogous lemma for not-necessarily-dense order

Reply:
I've renamed `le_of_forall_le_of_dense` instead